addons/mail/models/mail_message.py

```diff
@@ -1171,10 +1171,16 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
+        messages_per_partner = defaultdict(lambda: self.env['mail.message'])
+        for message in messages:
+            if not self.env.user._is_public():
+                messages_per_partner[self.env.user.partner_id] |= message
+            if message.author_id and not self.env.user._is_public():
```

Comment:
I meant checking `_is_public` for the `message.author_id` related user, not for the current user.

Reply:
Done